webrender/src/render_backend.rs

```diff
@@ -470,6 +481,38 @@ impl RenderBackend {
         let mut notifier = self.notifier.lock();
         notifier.as_mut().unwrap().as_mut().unwrap().new_scroll_frame_ready(composite_needed);
     }
+
+    fn handle_vr_compositor_command(&mut self, ctx_id: WebGLContextId, cmd: VRCompositorCommand) {
+        match cmd {
+            VRCompositorCommand::Create(compositor_id) => {
+                let mut creator = self.vr_compositor_creator.lock();
+                let compositor = creator.as_mut().unwrap().as_mut().unwrap().create_compositor(compositor_id);
+                if let Some(compositor) = compositor {
+                    self.vr_compositors.insert(compositor_id, compositor);
+                }
+            }
+            VRCompositorCommand::SyncPoses(compositor_id, near, far, sender) => {
+                if let Some(ref compositor) = self.vr_compositors.get(&compositor_id) {
+                    let pose = compositor.sync_poses(near, far);
+                    let _result = sender.send(Ok(pose));
+                } else {
+                    let _result = sender.send(Err(()));
+                }
+            }
+            VRCompositorCommand::SubmitFrame(compositor_id, left_bounds, right_bounds) => {
+                if let Some(ref compositor) = self.vr_compositors.get(&compositor_id) {
+                    let texture = self.resource_cache.get_webgl_texture(&ctx_id);
```

Comment:
> I understand this. But from WR's point of view, the user creates the Renderer, so the user is fully aware (and in control) of the thread that executes WebGL commands. 

The user creates the Renderer, but the Renderer internally creates the renderer_backend thread so the user has not really a direct access to that inner thread. At least I need a trait to run "hard code" because that thread can only can be accessed through predefined ipc-channel messages which have some limitations (can't send closures, complex objects, pointers, and more).

> Can't the user just manage the compositors (with their creator) then, only interacting with WR via the resource cache, if needed?
> 
> Disclaimer: I'm still learning the WR way of things, so I might just be missing the obvious answer.

Yes, I understand what you mean. I could reduce the current VRCompositor interfaces to a single vr_compositor_handler trait, that uses bytes (vec<u8> based messages) to forward-serialize all possible messages without defining them in WR. It has the pros of a less decoupled interface but it's a more obscure API. If that's the style you want in WR I can update the PR. I'm also interested in what @glennw thinks about both options.

The trait I'm talking about would look like this: 

```
// Single instance in render_backend
pub trait VRCompositorHandler {
    fn handle(&self, data: vec<u8>, sender:Option<IpcSender<vec<u8>>>);
}
```

```
// IPC-Channel message that uses the VRCompositorHandler instance
VRCompositorCommand(data: vec<u8>, sender:Option<IpcSender<vec<u8>>>)

```